configure

```diff
@@ -11490,7 +11490,7 @@ if ${ac_cv_lib_zstd_ZSTD_compressCCtx+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lzstd  $LIBS"
+LIBS="-l:libzstd.a  $LIBS"
```

Comment:
You should patch `configure.in` to regenerate configure script.

Reply:
Seems like `configure.in` just checks `--with-zstd` flag and if a library is available using `AC_CHECK_LIB`, but it doesn't generate a list to pass it to gcc

Reply:
`configure.in` is input script to generate `configure` with autotools. Please read documentation about `AC_CHECK_LIB` macro.

Reply:
I was able to use static linking in `configure.in`, but `configure` used dynamic linking anyway. As far as I understand, `configure.in` should form `LIBS` variable to be used by `configure`, but it's empty in the latter script and this script fills it by its own. As I see, we need to modify `configure as well`

Reply:
I'm not sure hot to use autoconf correctly. There's a macro `AC_CHECK_LIB` which check library presence on a machine, docs claim that it can check both static and dynamic libraries, but I couldn't find how can I specify which library I want. However, I can manually make it use static library in that way which replaces the default behavior and adds the library statically. 
```
AC_CHECK_LIB(zstd, ZSTD_compressCCtx, [
	LIBS="-l:libzstd.a $LIBS"
	AC_DEFINE([HAVE_LIBZSTD], [1], ["Define that zstd exists"])
  ],
```

But there's a problem that linking tests still use dynamic version of the library, i.e. test binary is linked against dynamic library and gpdb is linked against static library and I can't find a way to specify that behavior

Reply:
Added -l:libzstd to configure.in